vertical-pod-autoscaler/e2e/updater.go

```diff
@@ -78,18 +74,96 @@ var _ = UpdaterE2eDescribe("Updater", func() {
 
 		err = waitForPodSetChangedInDeployment(c, d, podList)
 		gomega.Expect(err).NotTo(gomega.HaveOccurred())
 
 	})
+
+	ginkgo.It("observes pod disruption budget", func() {
+
+		ginkgo.By("Setting up a hamster deployment")
+		c := f.ClientSet
+		ns := f.Namespace.Name
+
+		cpuQuantity := ParseQuantityOrDie("10m")
+		memoryQuantity := ParseQuantityOrDie("10Mi")
+
+		d := NewHamsterDeploymentWithResources(f, cpuQuantity, memoryQuantity)
+		replicas := int32(10)
+		d.Spec.Replicas = &replicas
+		d, err := c.AppsV1().Deployments(ns).Create(d)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		err = framework.WaitForDeploymentComplete(c, d)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		podList, err := framework.GetPodsForDeployment(c, d)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		podSet := makePodSet(podList)
+
+		ginkgo.By("Setting up prohibitive PDB for hamster deployment")
+		maxUnavailable := intstr.FromInt(0)
+		pdb := &policyv1beta1.PodDisruptionBudget{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "hamster-pdb",
+			},
+			Spec: policyv1beta1.PodDisruptionBudgetSpec{
+				MaxUnavailable: &maxUnavailable,
+				Selector: &metav1.LabelSelector{
+					MatchLabels: map[string]string{"app": "hamster"},
+				},
+			},
+		}
+		_, err = c.PolicyV1beta1().PodDisruptionBudgets(ns).Create(pdb)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		ginkgo.By("Setting up a VPA CRD")
+		vpaCRD := NewVPA(f, "hamster-vpa", &metav1.LabelSelector{
+			MatchLabels: map[string]string{
+				"app": "hamster",
+			},
+		})
+
+		newCPUQuantity := ParseQuantityOrDie("25m")
+		resourceList := apiv1.ResourceList{apiv1.ResourceCPU: newCPUQuantity}
+
+		vpaCRD.Status.Recommendation = &vpa_types.RecommendedPodResources{
+			ContainerRecommendations: []vpa_types.RecommendedContainerResources{{
+				ContainerName: "hamster",
+				Target:        resourceList,
+				LowerBound:    resourceList,
+				UpperBound:    resourceList,
+			}},
+		}
+		InstallVPA(f, vpaCRD)
+
+		ginkgo.By(fmt.Sprintf("Waiting for pods to be restarted, hoping it won't happen, sleep for %s", VpaEvictionTimeout.String()))
+		time.Sleep(VpaEvictionTimeout)
+		restarted, err := isPodSetChanged(c, d, podSet)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		gomega.Expect(restarted).To(gomega.BeFalse())
+
+		ginkgo.By("Updating the PDB to allow for multiple pods to be evicted")
+		maxUnavailable = intstr.FromInt(3)
+		newPdb := pdb.DeepCopy()
+		newPdb.Spec.MaxUnavailable = &maxUnavailable
+		newPdb.ObjectMeta.Name = "hamster-pdb-2"
+		// Creating new PDB and removing old one, since PDBs are immutable at the moment
+		_, err = c.PolicyV1beta1().PodDisruptionBudgets(ns).Create(newPdb)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		err = c.PolicyV1beta1().PodDisruptionBudgets(ns).Delete(pdb.Name, &metav1.DeleteOptions{})
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		ginkgo.By(fmt.Sprintf("Waiting for pods to be restarted, sleep for %s", VpaEvictionTimeout.String()))
```

Comment:
Is it possible to verify here that the Pod evictions were done in parallel? I think this is one of the goals of the test (using PDB > 1) so it would be nice to have this verified.

Reply:
OK. Maybe a one-line comment on the relation between this 7 and 3 minutes in time-out would be nice - we can see it now but it might be not so obvious in a month.

Reply:
Good point. Done